src/wavm/threads.cpp

```diff
@@ -286,83 +286,123 @@ I32 s__futex(I32 uaddrPtr,
     return returnValue;
 }
 
-/*
- * --------------------------
- * STUBBED PTHREADS
- * --------------------------
- */
+// --------------------------
+// PTHREAD MUTEXES - We support pthread mutexes locally as they're important to
+// support thread-safe libc operations
+// --------------------------
 
 WAVM_DEFINE_INTRINSIC_FUNCTION(env,
                                "pthread_mutex_init",
                                I32,
                                pthread_mutex_init,
-                               I32 a,
-                               I32 b)
+                               I32 mx,
```

Comment:
Note that we just use the integer wasm offset for the given mutex to identify it.